yescrypt/YescryptSettings.cs

```diff
@@ -100,7 +99,8 @@ public YescryptSettings(string encoded) : this()
 
             // Decode Salt
             reader = new B64StringReader(parts[3]);
-            salt = reader.ReadBytes(16);
+            var saltLen = (int)Math.Ceiling(parts[3].Length / 1.375);
+            salt = reader.ReadBytes(saltLen);

```

Comment:
`B64StringReader.GetBytes(int length)` will return the specified number of bytes _or the maximum number of bytes in the left in the b64 string_.  See:

https://github.com/petrsnm/yescrypt-net/blob/79dd4d6b6d294bb77b8ccb6677303c1dc488da66/yescrypt/B64StringReader.cs#L105

Therefore, it is not necessary to pre-calculate the saltLen.  The best fix would be to simply modify line 103 to be:

`salt = reader.ReadBytes(64) // read up to maximum of 512 bits of salt`

If we attempt to read 64 bytes (512 bits, the maximum that should exist based on `crypt` documentation) and there are only 18 bytes (192 bits) of salt left in the base64 string then all the remaining 18 bytes in the string will be returned.   

Can you test the above change.  If it works, please modify your PR to remove the `saltLen` arithmetic.

Reply:
Made the change to always read up to the maximum length of the salt, but it broke the tests as the last byte of `rval` was trimmed. Added a check to the part when the array gets resized, and a few extra test cases with different salt lengths. With the original calculated input size it wasn't an issue as `rvalIndex` was always equal to `rval.Length`, but not with the fixed size of 64 bytes - in cases where `bits % 24 == 0` resizing the array caused the loss of its last element.